drivers/i3c/i3c_dw.c

```diff
@@ -1617,10 +1635,18 @@ static void enable_interrupts(const struct device *dev)
 
 	sys_write32(INTR_ALL, config->regs + INTR_STATUS);
 
-	sys_write32(INTR_SLAVE_MASK | INTR_MASTER_MASK, config->regs + INTR_STATUS_EN);
-	sys_write32(INTR_SLAVE_MASK | INTR_MASTER_MASK, config->regs + INTR_SIGNAL_EN);
+	/* Enable interrupts */
+#if defined(CONFIG_I3C_CONTROLLER) && defined(CONFIG_I3C_TARGET)
+	intr_mask = INTR_MASTER_MASK | INTR_SLAVE_MASK;
+#elif defined(CONFIG_I3C_CONTROLLER)
+	intr_mask = INTR_MASTER_MASK;
+#elif defined(CONFIG_I3C_TARGET)
+	intr_mask = INTR_SLAVE_MASK;
+#endif
```

Comment:
nothing wrong with the lines here, just wondering if it becomes more "readable" in the format:

```c
#if defined(CONFIG_I3C_CONTROLLER) 
#if defined(CONFIG_I3C_TARGET)
	intr_mask = INTR_MASTER_MASK | INTR_SLAVE_MASK;
#else
	intr_mask = INTR_MASTER_MASK;
#endif /* CONFIG_I3C_TARGET */
#elif defined(CONFIG_I3C_TARGET)
	intr_mask = INTR_SLAVE_MASK;
#endif
```

Theory being it just groups the sections together.  Again, not a blocking item on the PR.

Reply:
umm... I think that is harder to read for me with that nested #if

Reply:
certainly a preference thing, hence the start with nothing wrong.